alica_engine/src/engine/modelmanagement/factories/Factory.cpp

```diff
@@ -45,28 +45,22 @@ int64_t Factory::getReferencedId(const std::string& idString)
-                essentials::FileSystem::endsWith(locator, alica::Strings::plantype_extension) ||
-                essentials::FileSystem::endsWith(locator, alica::Strings::condition_extension)) {
-            fileReferenced = essentials::FileSystem::combinePaths(modelManager->basePlanPath, locator);
-        } else if (essentials::FileSystem::endsWith(locator, alica::Strings::taskrepository_extension)) {
-            fileReferenced = essentials::FileSystem::combinePaths(modelManager->baseTaskPath, locator);
-        } else if (essentials::FileSystem::endsWith(locator, alica::Strings::roleset_extension)) {
-            fileReferenced = essentials::FileSystem::combinePaths(modelManager->baseRolePath, locator);
-        } else if (essentials::FileSystem::endsWith(locator, alica::Strings::configuration_extension)) {
-            fileReferenced = essentials::FileSystem::combinePaths(modelManager->basePlanPath, locator);
-        } else {
-            Logging::logDebug("Factory") << "Unknown file extension: " << locator;
-        }
-
-        if (std::find(std::begin(modelManager->filesParsed), std::end(modelManager->filesParsed), fileReferenced) == std::end(modelManager->filesParsed) &&
-                std::find(std::begin(modelManager->filesToParse), std::end(modelManager->filesToParse), fileReferenced) ==
-                        std::end(modelManager->filesToParse)) {
-            modelManager->filesToParse.push_back(fileReferenced);
+    std::string fileName = idString.substr(0, idxOfHashtag);
+    if (auto pathSeparatorPos = fileName.find_last_of(essentials::FileSystem::PATH_SEPARATOR); pathSeparatorPos != std::string::npos) {
+        Logging::logWarn(LOGNAME) << "File name expected, however path is specified, path: " << fileName << ", will only consider the file name instead";
+        fileName = fileName.substr(pathSeparatorPos + 1);
+    }
+    if (!fileName.empty()) {
+        if (!essentials::FileSystem::endsWith(fileName, alica::Strings::plan_extension) &&
+                !essentials::FileSystem::endsWith(fileName, alica::Strings::behaviour_extension) &&
+                !essentials::FileSystem::endsWith(fileName, alica::Strings::plantype_extension) &&
+                !essentials::FileSystem::endsWith(fileName, alica::Strings::condition_extension) &&
+                !essentials::FileSystem::endsWith(fileName, alica::Strings::taskrepository_extension) &&
+                !essentials::FileSystem::endsWith(fileName, alica::Strings::roleset_extension)) {
+            Logging::logWarn(LOGNAME) << "Unknown file extension: " << fileName;
+        } else if (std::find(std::begin(modelManager->filesParsed), std::end(modelManager->filesParsed), fileName) == std::end(modelManager->filesParsed) &&
+                   std::find(std::begin(modelManager->filesToParse), std::end(modelManager->filesToParse), fileName) == std::end(modelManager->filesToParse)) {
+            modelManager->filesToParse.push_back(fileName);
         }
```

Comment:
perhaps else branch also should be here

Reply:
else is not required, because there is nothing to do in it. Basically the else block will be hit when the file has a valid extension but is already either parsed or included in the list of files to be parsed. Therefore nothing needs to be done in either case.